pinot-broker/src/main/java/org/apache/pinot/broker/routing/RoutingManager.java

```diff
@@ -332,6 +334,7 @@ public synchronized void buildRouting(String tableNameWithType) {
         timeBoundaryManager = new TimeBoundaryManager(tableConfig, _propertyStore);
         timeBoundaryManager.init(externalView, onlineSegments);
       }
+      fetchSegmentsInErrorState(externalView, noReplicaSegments);
```

Comment:
You need to do this for realtime table also, since that may also have segments in ERROR state

Reply:
Added.